test/lint/commit-script-check.sh

```diff
@@ -17,6 +17,11 @@ if test -z "$1"; then
     exit 1
 fi
 
+if ! sed --help | grep -q 'GNU'; then
+    echo "Error: the installed sed package is not compatible. Please make sure you have GNU sed installed in your system.";
+    exit 1;
+fi
+
```

Comment:
> Heh, seems brittle...

Why don't check the required functionality directly:
```suggestion
echo "test" > tmp
sed -i 's/test/ok/' tmp &> /dev/null
ret=$?
rm tmp
if [ $ret -ne 0 ]; then
  echo "Error: the installed sed package is not compatible. Please make sure you have GNU sed installed in your system."
  exit 1
fi

```
?

Reply:
Hello @hebasto,

Thank you for the suggestion, I was thinking to use this approach in-fact, but I avoided it due to performance reasons, please correct me if I'm wrong, by running string substitution, will require to start PCRE engine before performing string replacement, also it has overhead of creating and removing temporary file.

Reply:
Correct. There are performance issues with this approach. But, as almost always, performance should not be considered at first place :)
I mean, are performance drawbacks a real issue?

Reply:
I see your point, if we have enough consent regarding this approach, I'll be happy to refactor the current changes.
Thank you again @hebasto
What do you think @shaavan, @katesalazar and @MarcoFalke ?

Reply:
I think all of these ways to do this same thing are good.
I could also concept NAK both this and the parent issue.
IDK who are we forcing this on, besides macOS users.
But I suppose macOS users should be OK to slip in their
stack just one more trusted build.